docs/graphql.md

```diff
@@ -0,0 +1,141 @@
+
+- `GraphQL:UserWithComments (/api/graphql)`
+
+This does not seem very common, so adding this is an opt-in option, like `graphql_postfix_path: true`.
+
+## Anonymous queries
+
+An Operation Name isn't required. When one isn't provided it's hard for us to tell apart the queries.
+
+1. Some clients generate `id`s from hashing the contents of the query (see [apollo-tooling](https://github.com/apollographql/apollo-tooling/blob/1dfd737eaf85b89b2cfb13913342e091e3c03d18/packages/apollo-codegen-core/src/compiler/visitors/generateOperationId.ts#L5)). This would split the anonymous queries into separate buckets.
+
+    A problem with this approach is that a user of the APM UI has no way to recognise queries in the transactions list before clicking through.
+
+    Using just the `id` will not reveal the true culprit since there can be variables associated with the query. Different values for the variables can lead to very different workloads and response times.
+
+2. Another approach is to simply label them `[anonymous]` or something similar.
+
+    A problem with _that_ approach is that the contents and thereby the relevant db queries and other sub-span actions that the server might do while resolving these queries may be wildly different making it hard to provide a _true_ sample waterfall.
+
+    These two examples for example will look the same for the top-level GraphQL spans but will represent significantly different workloads.
+
+    ```
+    [- anonymous graphql span --------------]
+      [- 1,000x SELECT * ---------------]
+        [- 1,000 more SELECT * -]
+
+    [- anonymous graphql span --------------]
+      [- SELECT id FROM users WHERE id=? -]
+    ```
+
+    We could consider _muting_ or ignoring all sub-spans to anonymous GraphQL queries and choose to rather show nothing than potentially wrong information.
+
+No one of these are perfect. Because the benefits of using `id`s in the worst case could be misleading anyway, we're going with option 2.
+
+To further help and nudge developers to use Operation Names for their queries, we could log a warning when coming across anonymous queries.
+
+```plain
+[DEBUG] Traced anonymous GraphQL query. Use Operation Names for your queries to better distinguish them in the APM UI.
+```
+
+This could also be a tooltip in the UI instead of a log message.
+
+Span name examples:
+- `GraphQL:ka8kadf8233kxcsc2929384384kdkdkc8383…`
+- `GraphQL:[anonymous]`
+
+## Batching/Multiplexing queries
+
+Some clients allow batching queries (see for example [apollo-link-batch-http](https://www.apollographql.com/docs/link/links/batch-http/#gatsby-focus-wrapper) or [dataloader](https://github.com/graphql/dataloader#batching).)
+
+So far it makes sense to update transaction names based on the span names. Essentially, in the best case, let the transactions be named after the Operation Names.
+
+However with multiple queries per HTTP request this wont work.
+
+Combining span names with `+` could work.
```

Comment:
Is there a limit on the number of queries that can be multiplexed? Do you have an idea of what's typical? If it's just a few then this sounds fine, but if there are even tens of them, then it would start getting pretty unwieldy.

Reply:
Yeah I had similar concerns as well https://github.com/elastic/apm/pull/208#discussion_r377139212

Reply:
There are no limit AFAIK for multiplexing. Most of the batching is based on number of calls that happens in same tick (Node.js context). But the rest is up to the batching library.

Reply:
You could imagine someone requesting tens or hundreds of individual records, each from its own query? I'm not sure that's _good_ GraphQL but I could definitely see someone do it.

We could max the `+` concatted names at perhaps 3 and flip it to `[multi]` for 4 or more queries in the same request?

Reply:
I like the max concatenation idea -- similar to how we truncate large local variables when we collect them in the python agent.